app/assets/javascripts/discourse/app/static/prosemirror/extensions/hashtag.js

```diff
@@ -0,0 +1,80 @@
+import { isBoundary } from "discourse/static/prosemirror/lib/markdown-it";
+
+/** @type {RichEditorExtension} */
+const extension = {
+  nodeSpec: {
+    hashtag: {
+      attrs: { name: {} },
+      inline: true,
+      group: "inline",
+      content: "text*",
+      atom: true,
+      draggable: true,
+      selectable: false,
+      parseDOM: [
+        {
+          tag: "a.hashtag-cooked",
+          preserveWhitespace: "full",
+          getAttrs: (dom) => {
+            return { name: dom.getAttribute("data-name") };
+          },
+        },
+      ],
+      toDOM: (node) => {
+        return [
+          "a",
+          { class: "hashtag-cooked", "data-name": node.attrs.name },
+          `#${node.attrs.name}`,
+        ];
+      },
+    },
+  },
+
+  inputRules: [
+    {
+      match: /(?<=^|\W)#([\u00C0-\u1FFF\u2C00-\uD7FF\w:-]{1,101})\s$/,
```

Comment:
This regex contains a positive lookbehind.

This is not supported in Safari < 16.4 and can cause the application to crash (see ee3ccc5b)

We currently support Safari 15.7+

Reply:
Thanks, updated!